### PR TITLE
Switch to systemd-detect-virt to detect virtualization environments

### PR DESCRIPTION
**Pull request by felixsch**
card: https://trello.com/c/BoIKyB91/3343-tedd-correctly-detect-hypervisors-virtualization

Moving from `lscpu` to `systemd-detect-virt` will allow SUSEConnect to detect many more different hypervisors and at the same time streamline data sent to SCC

**How to test this pull request:**

```
## No virtual machine

$ cd <connect>
$ make build

$ docker run --network=host --rm --privileged -ti -v $(pwd):/connect registry.suse.com/suse/sle15:15.6 /bin/bash
> cd /connect
> ./out/suseconnect -r <regcode>
> ./out/suseconnect --debug --keepalive
# expect: See that hypervisor is sent to empty string in t

## Virtualized environment (kvm)

# building suseconnect for sle15.4
$ cd <connect>

$ docker run --network=host --rm --privileged -ti -v $(pwd):/connect registry.suse.com/suse/sle15:15.4 /bin/bash
> zypper in -y make go1.21
> cd /connect
> make build
> exit

# copy it to the test host
$ cd <connect>
$ rsync -av out/suseconnect sles@<ip in the card>:
$ ssh sles@<ip in the card>
> ./suseconnect --debug --keepalive
# expect: See hypervisor ist set to `kvm``
```

Thank you for the review! :rocket:

Felix

### CODE REVIEW

File: internal/collectors/virtualization.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	if err != nil {
+		return NoResult, err
+	}
+
+	if string(output) == "none" {
+		return Result{"hypervisor": nil}, nil
+	}
+
+	// Historicly SCC API expects virtualization information with the key `hypervisor`. Use it!
```
Comment (ngetahun) [inline internal/collectors/virtualization.go:31]:
```suggestion
	// Historically SCC API expects virtualization information with the key `hypervisor`. Use it!
```

**Review by ngetahun — CHANGES_REQUESTED** [review]

**Review by ngetahun — COMMENTED** [review]
Please remove internal IPs from public PRs

**Comment by felixsch** [comment]
> Please remove internal IPs from public PRs

I moved it

File: internal/collectors/virtualization.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+type Virtualization struct{}
+
+func (Virtualization) run(arch Architecture) (Result, error) {
+
+	// We utilize systemd here to fetch the virtualization information. Since we do not hard require systemd
+	// there might be the possibility to run on a system without systemd installed (e.g. containers).
+	// This tool can detect a lot different hypervisors:
+	// https://github.com/systemd/systemd/blob/main/src/basic/virt.c#L1046
+	if !util.ExecutableExists("systemd-detect-virt") {
+		return NoResult, fmt.Errorf("can not detect virtualization: `system-detect-virt` executable not found")
```
Comment (lcaparroz) [inline internal/collectors/virtualization.go:18]: Small typo here:

```suggestion
		return NoResult, fmt.Errorf("can not detect virtualization: `systemd-detect-virt` executable not found")
```

By the way, do you think it's worth having `systemd-detect-virt` in a constant? It is used in at least three different spots.

Comment (felixsch) [reply internal/collectors/virtualization.go:18]: I have no opinion but I will move it to a constant :)

File: internal/collectors/virtualization.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+	"github.com/SUSE/connect-ng/internal/util"
+)
+
+type Virtualization struct{}
+
+func (Virtualization) run(arch Architecture) (Result, error) {
+
+	// We utilize systemd here to fetch the virtualization information. Since we do not hard require systemd
+	// there might be the possibility to run on a system without systemd installed (e.g. containers).
```
Comment (mssola) [inline internal/collectors/virtualization.go:14]: On that regard, could we consider detecting container engines as well? (see [this package](https://github.com/genuinetools/bpfd/tree/master/proc) which is being used by tools like [amicontained](https://github.com/genuinetools/amicontained).

**Review by ngetahun — APPROVED** [review]

File: internal/collectors/virtualization.go (hunk 3 above)
Comment (felixsch) [reply internal/collectors/virtualization.go:14]: We have this in our next epic CAT see our Roadmap https://trello.com/c/v0ae4HG6/450-cat-collect-additional-telemetry-data

File: internal/collectors/virtualization.go (hunk 2 above)
Comment (felixsch) [reply internal/collectors/virtualization.go:20]: moved it!

**Pull request merged**